jenkins/prepare-custom-leonardo-jupyter-dataproc-image.sh

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env bash
+
+set -e -x
+
+#
+# This script sets up a custom Dataproc image for Leonardo clusters.
+# See: https://cloud.google.com/dataproc/docs/guides/dataproc-images
+#
+
+#
+# Constants and Global Vars
+#
+leonardo_jupyter_docker_image_tag="dev"
```

Comment:
This script is used to actually create the custom Dataproc image. It is not currently hooked up to anything; but we intend to wire it to Jenkins once we do https://broadworkbench.atlassian.net/browse/IA-1265. But I wanted to commit the current version anyway so we can iterate on it.